examples/bluetooth/ble_bonding_peripheral.py

```diff
@@ -0,0 +1,193 @@
+        self._ble.config(mitm=True)
+        self._ble.config(io=_IO_CAPABILITY_DISPLAY_YESNO)
+        self._ble.active(True)
+        self._ble.config(addr_mode=2)
+        ((self._handle,),) = self._ble.gatts_register_services((_ENV_SENSE_SERVICE,))
+        self._connections = set()
+        self._payload = advertising_payload(
+            name=name, services=[_ENV_SENSE_UUID], appearance=_ADV_APPEARANCE_GENERIC_THERMOMETER
+        )
+        self._advertise()
+
+    def _irq(self, event, data):
+        # Track connections so we can send notifications.
+        if event == _IRQ_CENTRAL_CONNECT:
+            conn_handle, _, _ = data
+            self._connections.add(conn_handle)
+        elif event == _IRQ_CENTRAL_DISCONNECT:
+            conn_handle, _, _ = data
+            self._connections.remove(conn_handle)
+            self._save_secrets()
+            # Start advertising again to allow a new connection.
+            self._advertise()
+        elif event == _IRQ_ENCRYPTION_UPDATE:
+            conn_handle, encrypted, authenticated, bonded, key_size = data
+            print("encryption update", conn_handle, encrypted, authenticated, bonded, key_size)
+        elif event == _IRQ_PASSKEY_ACTION:
+            conn_handle, action, passkey = data
+            print("passkey action", conn_handle, action, passkey)
+            if action == _PASSKEY_ACTION_NUMCMP:
+                accept = int(input("accept? "))
+                self._ble.gap_passkey(conn_handle, action, accept)
+            elif action == _PASSKEY_ACTION_DISP:
+                print("displaying 123456")
+                self._ble.gap_passkey(conn_handle, action, 123456)
+            elif action == _PASSKEY_ACTION_INPUT:
+                print("prompting for passkey")
+                passkey = int(input("passkey? "))
+                self._ble.gap_passkey(conn_handle, action, passkey)
+            else:
+                print("unknown action")
+        elif event == _IRQ_GATTS_INDICATE_DONE:
+            conn_handle, value_handle, status = data
+        elif event == _IRQ_SET_SECRET:
+            sec_type, key, value = data
+            key = sec_type, bytes(key)
+            value = bytes(value) if value else None
+            print("set secret:", key, value)
+            if value is None:
+                if key in self._secrets:
+                    del self._secrets[key]
+                    return True
+                else:
+                    return False
+            else:
+                self._secrets[key] = value
+            return True
+        elif event == _IRQ_GET_SECRET:
+            sec_type, index, key = data
+            print("get secret:", sec_type, index, bytes(key) if key else None)
+            i = 0
+            if key is None:
+                for (t, _key), value in self._secrets.items():
+                    if t == sec_type:
+                        if i == index:
+                            return value
```

Comment:
```suggestion
                        if i == index:
                            return value
                        i++                            
```

Reply:
@jimmo This fixes the "samsung not working over restart" issue I was having, turns out it's got nothing to do with the public/random address thing

Reply:
yeah, I saw this as well

Reply:
this is now fixed in #6662